In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df_raw = pd.read_csv("2018 NCWIT Award for Aspirations in Computing Recipient.csv")

## Question 3

In [41]:
q3 = pd.read_csv('Question3Code.csv')

In [42]:
q3 = q3.dropna()
q3.head()

Respondent ID If you did not attend an event, please let us know why. CODE
0     6928372992           I don't think that I was supposed to go.        dk
1     6921763051  I wasn't recognized until after the events hap...        ra
3     6920022402                                 I was not asked to        na
4     6918387583                  There was not enough room for me.         p
5     6918090577               I did not know they were occurring.         dk

In [43]:
q3['CODE'].unique()

array(['dk', 'ra', 'na', 'p', 'con', 'nq', 'tp', 'ni'], dtype=object)

In [44]:
q3_gb = q3.groupby('CODE')
agg_d = {'CODE' : 'count'}
q3_agg = q3_gb.agg(agg_d)
q3_agg

CODE
CODE      
con    376
dk      55
na      46
ni       9
nq      38
p       28
ra       8
tp      62

## Question 5

In [57]:
df_gb_5 = df_raw.groupby('Have you joined the NCWIT Aspirations in Computing Facebook group?')
agg_d = {'Have you joined the NCWIT Aspirations in Computing Facebook group?' : 'count'}
df_agg_5 = df_gb_5.agg(agg_d)
df_agg_5

Have you joined the NCWIT Aspirations in Computing Facebook group?
Have you joined the NCWIT Aspirations in Comput...                                                                    
No                                                                                                976                 
Response                                                                                            1                 
Yes                                                                                               666

## Question 6

In [56]:
q6 = pd.read_csv('Question6Code.csv')

In [53]:
q6 = q6.dropna()
q6.head()

Respondent ID  \
1   6.928373e+09   
2   6.921763e+09   
5   6.918388e+09   
6   6.918091e+09   
7   6.917209e+09   

  In order to improve our program, please tell us why you have not joined the Facebook Group.  \
1  I did not know that there was a Facebook group...                                            
2  I actually did not have a Facebook at the time...                                            
5  I do not use Facebook. Most people my age have...                                            
6                  I do not have a Facebook account.                                            
7  I have not joined the Facebook group because I...                                            

  CODE  
1   dk  
2   nf  
5   nf  
6   nf  
7   nf

In [54]:
q6['CODE'].unique()

array(['dk', 'nf', 'na', 'p', 'ni', 'fp'], dtype=object)

In [55]:
q6_gb = q6.groupby('CODE')
agg_d = {'CODE' : 'count'}
q6_agg = q6_gb.agg(agg_d)
q6_agg

CODE
CODE      
dk     200
fp      45
na      48
nf     476
ni      16
p       99

## Question 7

In [37]:
q7 = pd.read_csv("Question7Code.csv")

In [38]:
q7 = q7.dropna()
q7.head()

Respondent ID  \
0   6.928373e+09   
1   6.921763e+09   
2   6.921763e+09   
4   6.920022e+09   
5   6.918388e+09   

  What is the best way for NCWIT to communicate opportunities with you? CODE  
0                                      Through email                       e  
1       Either through email or the Facebook group.                        e  
2       Either through email or the Facebook group.                       fb  
4                                              Email                       e  
5                    They can email my school email.                       e

In [39]:
q7['CODE'].unique()

array(['e', 'fb', 't', 'tw', 'gm', 'web', 'p', 'news', 'sm', 'i', 's'],
      dtype=object)

In [40]:
q7_gb = q7.groupby('CODE')
agg_d = {'CODE' : 'count'}
q7_agg = q7_gb.agg(agg_d)
q7_agg

CODE
CODE      
e     1352
fb     187
gm       6
i       13
news    18
p       21
s       11
sm      24
t       73
tw       6
web     11

## Question 8

In [29]:
q8 = pd.read_csv("Question8Code.csv")

In [30]:
q8 = q8.dropna()
q8.head()

Respondent ID  \
0     6928372992   
1     6921763051   
3     6920022402   
4     6920022402   
5     6920022402   

  In addition to Facebook, what would be the best way(s) for you to connect with the rest of the Aspirations community?  \
0   A specific website or page with a way to connect                                                                      
1  I'm not sure what the Facebook has, but I find...                                                                      
3                        Snapchat, instagram, emails                                                                      
4                        Snapchat, instagram, emails                                                                      
5                        Snapchat, instagram, emails                                                                      

  CODE  
0  web  
1    e  
3    e  
4   sn  
5    i

In [31]:
q8['CODE'].unique()

array(['web', 'e', 'sn', 'i', 'tw', 'sm', 'fm', 'd', 'gm', 'goo', 'ev',
       'fb', 'li', 't', 'p', 'news', 'o'], dtype=object)

In [32]:
q8_gb = q8.groupby('CODE')
agg_d = {'CODE' : 'count'}
q8_agg = q8_gb.agg(agg_d)
q8_agg

CODE
CODE      
d       30
e      334
ev     136
fb      96
fm      21
gm     172
goo     25
i      303
li      49
news    10
o       20
p       12
sm     104
sn      73
t       47
tw     114
web     52

## Question 11

In [3]:
q11 = df_raw[["Please rate the following Impact items from Strongly Agree to Strongly Disagree.Being a recipient of the Award for Aspirations in Computing...",
                    'Unnamed: 23', 'Unnamed: 24', 'Unnamed: 25', 'Unnamed: 26', 'Unnamed: 27', 'Unnamed: 28',
                    "Unnamed: 29"]].dropna().replace("Strongly Disagree", 1).replace("Disagree", 2).replace("Neither Agree nor Disagree", 3).replace("Agree", 4).replace("Strongly Agree", 5)

In [4]:
q11 = pd.read_csv("Question11.csv")
q11 = q11[["Increased my commitment to pursuing a computer science degree in college.", 
     "Increased my confidence in my ability to pursue a technology career.", 
    "Positively impacted my self-esteem.", 'Helped me see myself as a "computing person."', 
    "Made me feel part of a community of women in computing.", 
     "Fed my interest in a computing career.", "Inspired me to emulate the successful women I saw at the event ceremony.",
    "Motivated me to stay in touch with the young women I met at the event ceremony."]].dropna().replace("Strongly Disagree", 1).replace("Disagree", 2).replace("Neither Agree nor Disagree", 3).replace("Agree", 4).replace("Strongly Agree", 5)
q11.head()

Increased my commitment to pursuing a computer science degree in college.  \
0                                                   4                           
1                                                   4                           
5                                                   4                           
8                                                   3                           
11                                                  4                           

    Increased my confidence in my ability to pursue a technology career.  \
0                                                   4                      
1                                                   5                      
5                                                   4                      
8                                                   3                      
11                                                  4                      

    Positively impacted my self-esteem.  \
0                                     4   
1                                     4   
5                                     4   
8                                     3   
11                                    4   

    Helped me see myself as a "computing person."  \
0                                               4   
1                                               5   
5                                               4   
8                                               2   
11                                              4   

    Made me feel part of a community of women in computing.  \
0                                                   4         
1                                                   5         
5                                                   5         
8                                                   4         
11                                                  4         

    Fed my interest in a computing career.  \
0                                        4   
1                                        5   
5                                        5   
8                                        3   
11                                       4   

    Inspired me to emulate the successful women I saw at the event ceremony.  \
0                                                   3                          
1                                                   3                          
5                                                   3                          
8                                                   3                          
11                                                  4                          

    Motivated me to stay in touch with the young women I met at the event ceremony.  
0                                                   3                                
1                                                   3                                
5                                                   3                                
8                                                   3                                
11                                                  3

In [5]:
q11.mean()

Increased my commitment to pursuing a computer science degree in college.          3.951737
Increased my confidence in my ability to pursue a technology career.               4.158301
Positively impacted my self-esteem.                                                4.248069
Helped me see myself as a "computing person."                                      4.025097
Made me feel part of a community of women in computing.                            4.250965
Fed my interest in a computing career.                                             4.014479
Inspired me to emulate the successful women I saw at the event ceremony.           4.005792
Motivated me to stay in touch with the young women I met at the event ceremony.    3.476834
dtype: float64

In [6]:
q11.std()

Increased my commitment to pursuing a computer science degree in college.          0.963882
Increased my confidence in my ability to pursue a technology career.               0.905186
Positively impacted my self-esteem.                                                0.890772
Helped me see myself as a "computing person."                                      0.945545
Made me feel part of a community of women in computing.                            0.923525
Fed my interest in a computing career.                                             0.939104
Inspired me to emulate the successful women I saw at the event ceremony.           1.012943
Motivated me to stay in touch with the young women I met at the event ceremony.    1.070665
dtype: float64

In [16]:
from scipy.stats import pearsonr
# help(pearsonr)

x = (["Increased my commitment to pursuing a computer science degree in college.",
     "Increased my confidence in my ability to pursue a technology career.",
    "Positively impacted my self-esteem.", 'Helped me see myself as a "computing person."',
    "Made me feel part of a community of women in computing.",
     "Fed my interest in a computing career.", "Inspired me to emulate the successful women I saw at the event ceremony.",
    "Motivated me to stay in touch with the young women I met at the event ceremony."])

for thing in x:
    pr = pearsonr(q11[thing], q11['Motivated me to stay in touch with the young women I met at the event ceremony.'])
    print(pr)

(0.46983877638836996, 5.230087272179904e-58)
(0.4723489026788479, 1.0807729730327622e-58)
(0.4381060893699225, 7.951385985206021e-50)
(0.4682238631434008, 1.432390944408198e-57)
(0.5179079782472549, 3.64134248893318e-72)
(0.5283660371166555, 1.4789066526694268e-75)
(0.5996888479407988, 3.4823558999573285e-102)
(1.0, 0.0)


Correlation for each part of Q11:

1. Increased my commitment to pursuing a computer science degree in college.
(1.0, 0.0)
(0.7019868752563281, 1.3547885130046411e-154)
(0.5608543960857546, 6.9468434168406375e-87)
(0.6660228740075038, 8.73300049437047e-134)
(0.5758534963810277, 1.4956390328247555e-92)
(0.7607524408132378, 2.5305395586964962e-196)
(0.5613779973320903, 4.454729820042528e-87)
(0.46983877638836996, 5.230087272179904e-58)

2. Increased my confidence in my ability to pursue a technology career.
(0.7019868752563281, 1.3547885130046411e-154)
(1.0, 0.0)
(0.6977738492831663, 5.369838698315858e-152)
(0.6873444727765234, 9.342285273617402e-146)
(0.6250916963779435, 2.1932847422274703e-113)
(0.7144970515105648, 1.37894916056079e-162)
(0.601743373986242, 4.712509792137798e-103)
(0.4723489026788479, 1.0807729730327622e-58)

3. Positively impacted my self-esteem.
(0.5608543960857546, 6.9468434168406375e-87)
(0.6977738492831663, 5.369838698315858e-152)
(1.0, 0.0)
(0.6659638842879294, 9.395208442553363e-134)
(0.612492287412769, 1.0568249233289392e-107)
(0.5870589747590875, 5.569634436148925e-97)
(0.5819919538006707, 5.887224774605301e-95)
(0.4381060893699225, 7.951385985206021e-50)

4. Helped me see myself as a "computing person."
(0.6660228740075038, 8.73300049437047e-134)
(0.6873444727765234, 9.342285273617402e-146)
(0.6659638842879294, 9.395208442553363e-134)
(1.0, 0.0)
(0.6345171510035218, 8.243538441263483e-118)
(0.7003187817642246, 1.465585850045276e-153)
(0.5637508716252908, 5.888398501486674e-88)
(0.4682238631434008, 1.432390944408198e-57)

5. Made me feel part of a community of women in computing.
(0.5758534963810277, 1.4956390328247555e-92)
(0.6250916963779435, 2.1932847422274703e-113)
(0.612492287412769, 1.0568249233289392e-107)
(0.6345171510035218, 8.243538441263483e-118)
(1.0, 0.0)
(0.6519710682832518, 2.0052863394257116e-126)
(0.5995482123657097, 3.991180015077186e-102)
(0.5179079782472549, 3.64134248893318e-72)

6. Fed my interest in a computing career.
(0.7607524408132378, 2.5305395586964962e-196)
(0.7144970515105648, 1.37894916056079e-162)
(0.5870589747590875, 5.569634436148925e-97)
(0.7003187817642246, 1.465585850045276e-153)
(0.6519710682832518, 2.0052863394257116e-126)
(1.0, 0.0)
(0.6123727524894432, 1.1931433426933272e-107)
(0.5283660371166555, 1.4789066526694268e-75)

7. Inspired me to emulate the successful women I saw at the event ceremony.
(0.5613779973320903, 4.454729820042528e-87)
(0.601743373986242, 4.712509792137798e-103)
(0.5819919538006707, 5.887224774605301e-95)
(0.5637508716252908, 5.888398501486674e-88)
(0.5995482123657097, 3.991180015077186e-102)
(0.6123727524894432, 1.1931433426933272e-107)
(1.0, 0.0)
(0.5996888479407988, 3.4823558999573285e-102)

8. Motivated me to stay in touch with the young women I met at the event ceremony.
(0.46983877638836996, 5.230087272179904e-58)
(0.4723489026788479, 1.0807729730327622e-58)
(0.4381060893699225, 7.951385985206021e-50)
(0.4682238631434008, 1.432390944408198e-57)
(0.5179079782472549, 3.64134248893318e-72)
(0.5283660371166555, 1.4789066526694268e-75)
(0.5996888479407988, 3.4823558999573285e-102)
(1.0, 0.0)